scripts/generate-github-summary.sh

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+set -e
+
+REPORT_FILE="$1"
+OUTPUT_FILE="$2"
+
+ICON_TESTS="📝"
+ICON_SUCCESS="✅"
+ICON_FAIL="❌"
+ICON_SKIP="⏭️"
+ICON_FLAKY="🍂"
+ICON_TIME="⏱️"
+
+TOTAL=$(jq '.results.summary.tests // 0' "$REPORT_FILE")
+PASSED=$(jq '.results.summary.passed // 0' "$REPORT_FILE")
+FAILED=$(jq '.results.summary.failed // 0' "$REPORT_FILE")
+SKIPPED=$(jq '.results.summary.skipped // 0' "$REPORT_FILE")
+START=$(jq '.results.summary.start // 0' "$REPORT_FILE")
+STOP=$(jq '.results.summary.stop // 0' "$REPORT_FILE")
+FLAKY_COUNT=$(jq '[.results.tests[] | select(.flaky == true)] | length' "$REPORT_FILE")
+
+DURATION_SEC=$(( (STOP - START) / 1000 ))
+DURATION=$(printf '%02d:%02d:%02d' $((DURATION_SEC/3600)) $((DURATION_SEC%3600/60)) $((DURATION_SEC%60)))
+
+REPO="${GITHUB_REPOSITORY}"
+RUN_ID="${GITHUB_RUN_ID}"
+TOKEN="${GITHUB_TOKEN}"
+
+ARTIFACTS=$(curl -s -H "Authorization: Bearer $TOKEN" \
+  -H "Accept: application/vnd.github+json" \
+  "https://api.github.com/repos/${REPO}/actions/runs/${RUN_ID}/artifacts")
```

Comment:
I believe we can assume gh is installed:
```suggestion
ARTIFACTS=$(gh api repos/${REPO}/actions/runs/${RUN_ID}/artifacts)
```